src/devices/Bno055/Bno055Sensor.cs

```diff
@@ -123,24 +123,47 @@ public Bno055Sensor(I2cDevice i2cDevice, OperationMode operationMode = Operation
             Info.FirmwareVersion = new Version(ReadByte(Registers.SW_REV_ID_MSB), ReadByte(Registers.SW_REV_ID_LSB));
             Info.BootloaderVersion = new Version(ReadByte(Registers.BL_REV_ID), 0);
 
-            // Check if a reset is needed
-            if (TemperatureSource != TemperatureSource.Gyroscope)
+            _operationMode = operationMode;
+            initializeRegisters();
+        }
+
+        private static readonly byte[][] s_registerDefaults =
+        {
+            new byte[] { (byte)Registers.UNIT_SEL, (byte)(
+                Units.AccelerationMeterPerSecond |
+                Units.AngularRateDegreePerSecond |
+                Units.DataOutputFormatWindows |
+                Units.EulerAnglesDegrees |
+                Units.TemperatureCelsius
+            ) },
+            new byte[] { (byte)Registers.TEMP_SOURCE, (byte)TemperatureSource.Gyroscope },
+            new byte[] { (byte)Registers.PWR_MODE, (byte)PowerMode.Normal },
+            new byte[] { (byte)Registers.AXIS_MAP_CONFIG,
+                0x24, // AXIS_MAP_CONFIG
+                0, // AXIS_MAP_SIGN
+            },
+            new byte[] { (byte)Registers.ACCEL_OFFSET_X_LSB,
+                0, 0, 0, 0, 0, 0, // ACC_OFFSET_*_*SB
+                0, 0, 0, 0, 0, 0, // MAC_OFFSET_*_*SB
+                0, 0, 0, 0, 0, 0, // GYR_OFFSET_*_*SB
+                0, 0, 0, 0, // *_RADIUS_*SB
+            }
+        };
+
+        private void initializeRegisters()
```

Comment:
```suggestion
        private void InitializeRegisters()
```

Sorry, didn't notice earlier

Reply:
Don't worry about it.
Is there a reason the code style rules are not tracked as errors? Since you are going to enforce them via code review anyways… All my projects do that and I pretty much didn't notice the smudge.

Reply:
I guess we haven't enabled them yet. I will create an issue on that.

Reply:
https://github.com/dotnet/iot/issues/816